numpy/linalg/tests/test_linalg.py

```diff
@@ -1582,9 +1582,28 @@ def check_qr(self, a):
         assert_(isinstance(r2, a_type))
         assert_almost_equal(r2, r1)
 
-    def test_qr_empty(self):
-        a = np.zeros((0, 2))
-        assert_raises(linalg.LinAlgError, linalg.qr, a)
+
+    @pytest.mark.parametrize(["m", "n"], [
+        (0, 4, 1),
+        (0, 4, 2),
+        (4, 0, 1),
+        (4, 0, 2),
+        (4, 2, 0),
+        (0, 0, 0)
```

Comment:
Theses are triplets, but you only have two variables. I assume you just want `[(4, 0), (0, 4), (0, 0)]`?

Reply:
oh shit...

Reply:
thanks